File: benchmark/prime_sieve.js
// Prime Sieve - count primes under 10000

function isPrime(n) {
    if (n < 2) return false;
    for (let i = 2; i * i <= n; i++) {
        if (n % i === 0) return false;
    }
    return true;
}

let count = 0;
for (let i = 2; i < 10000; i++) {
    if (isPrime(i)) count++;
}

console.log(count);

File: benchmark/prime_sieve.py
"""Prime Sieve - count primes under 10000"""

def is_prime(n):
    if n < 2:
        return False
    i = 2
    while i * i <= n:
        if n % i == 0:
            return False
        i += 1
    return True

count = 0
for i in range(2, 10000):
    if is_prime(i):
        count += 1

print(count)
